src/netex_idf/offers.rs

```diff
@@ -267,13 +283,105 @@ fn enhance_with_object_code(
     enhanced_routes
 }
 
+fn arrival_departure_times(el: &Element) -> Result<(Time, Time)> {
+    fn time(el: &Element, node_name: &str) -> Result<Time> {
+        Ok(el.try_only_child(node_name)?.text().parse()?)
+    }
+    let offset: u32 = match el.try_only_child("DepartureDayOffset")?.text().parse() {
+        Ok(offset) if offset > 0 => offset,
+        Ok(offset) if offset == 0 => 0,
+        _ => 0,
+    };
+    let offset_time = Time::new(24 * offset, 0, 0);
+    let arrival_time = time(el, "ArrivalTime")?;
+    let departure_time = time(el, "DepartureTime")?;
+
+    Ok((arrival_time + offset_time, departure_time + offset_time))
+}
+
+fn boarding_type(el: &Element, node_name: &str) -> u8 {
+    el.only_child(node_name)
+        .map(|node| (node.text() == "false") as u8)
+        .unwrap_or(0)
+}
+
+fn stop_point_idx(
+    sp_in_jp: &Element,
+    stop_points: &CollectionWithId<StopPoint>,
+    map_schedule_stop_point_quay: &HashMap<String, String>,
+) -> Result<Idx<StopPoint>> {
+    sp_in_jp
+        .try_only_child("ScheduledStopPointRef")
+        .and_then(|ssp_ref_el| ssp_ref_el.try_attribute::<String>("ref"))
+        .and_then(|ssp_ref| {
+            map_schedule_stop_point_quay.get(&ssp_ref).ok_or_else(|| {
+                format_err!(
+                    "QuayRef corresponding to ScheduledStopPointRef {} not found",
+                    ssp_ref
+                )
+            })
+        })
+        .and_then(|quay_ref| {
+            stop_points
+                .get_idx(quay_ref)
+                .ok_or_else(|| format_err!("stop point {} not found", quay_ref))
+        })
+}
+
+fn stop_times(
+    service_journey_element: &Element,
+    map_journeypatterns: &HashMap<String, &Element>,
+    stop_points: &CollectionWithId<StopPoint>,
+    map_schedule_stop_point_quay: &HashMap<String, String>,
+) -> Result<Vec<StopTime>> {
+    let sj_el = service_journey_element.only_child("passingTimes");
```

Comment:
Maybe you don't need this variable?
```rust
let timetable_passing_times = service_journey_element
    .only_child("passingTimes")
    .into_iter()
    .flat_map(|el| el.children());
```

Reply:
It's working just fine for me. Note that I'm using `.into_iter()`, not `.iter()`.